example/index.html

```diff
@@ -1,23 +1,26 @@
 ---
-layout: default
+layout: page
 ---
 
 <div class="home">
 
   <h1 class="page-heading">Posts</h1>
 
+  {% assign custom_url = site.url | append: site.baseurl %}
+  {% assign full_base_url = site.github.url | default: custom_url %}
```

Comment:
We need to assume users are going to look to the default theme as an example of how they should build their own site... I'd really rather avoid this complexity if possible. Is there any way to do this without assigns?